### ISSUE TITLE
How this will work with formbuiler and form array?

### ISSUE THREAD

**Issue by outsourcegb** [yuyang041060120/ng2-validation#80]
How gt will work with form array where one value should be greater then other but in form array ?

Ex.
this.sampleForm = fb.group({
      'name': ['', Validators.required],
      'nested_attributes': fb.array([
        fb.group({
          field_min: ['', [Validators.required, CustomValidators.min(2)]],
          field_max: ['', [Validators.required, CustomValidators.gt(field_max)]]
        })
      ])
    ])

**Comment by crutchcorn** [comment]
I too would love to know how to get around this. I am trying to do this with a password confirmation. I was trying to trick the system by having a variable `confirmPass` and doing something like 
`field_max: ['', [Validators.required, CustomValidators.equalTo(this.confirmPass)]]`, but I then get the error `_this.subscribe is not a function`.

**Comment by guilhermejcgois** [comment]
Do you have tried to use [FormBuild.control](https://angular.io/docs/ts/latest/api/forms/index/FormBuilder-class.html#!#control-anchor)?

For example:

```
  fb.group({
    'comp1': fb.control({}, []),
    'comp2': fb.control({}, []),
    ...
    'compN': fb.control({}, [])
  });
```

**Comment by se-aurasys** [comment]
@guilhermejcgois , no. But still I doubt how they will work with array. Tho, will update once I check it.

**Comment by se-aurasys** [comment]
@guilhermejcgois ,

Tried something like this 
```
this.formName = fb.group({
  ArrayForm = this.fb.array([])
});
```
  
```
setFormArray() {
  const minValField = new FormControl('', Validators.required);
  const maxValField = new FormControl('', [Validators.required, 
  CustomValidators.gt(minValField)]);
  
  return this.fb.group({
      sampleObject: this.fb.group({
        min_field: minValField,
        max_field: maxValField,
      })
    })
};
```

but no luck.
*It works for only equalTo but no other validators like min and gt 
any idea?

**Comment by se-aurasys** [comment]
I just created new gte validator that can handle this


```
import { AbstractControl, ValidatorFn } from '@angular/forms';

export const newGte = (gtControl: AbstractControl): ValidatorFn => {
  let subscribe: boolean = false;

  return (control: AbstractControl): { [key: string]: boolean } => {
    if (!subscribe) {
      subscribe = true;
      gtControl.valueChanges.subscribe(() => {
        control.updateValueAndValidity();
      });
    }

    let v: string = control.value;

    return v >= gtControl.value ? null : { newGte: true };
  };
};
```

**Comment by guilhermejcgois** [comment]
That is strange... IDK, how are you using your newGte on FormArray? I'll give a look when I've free time, do you have some plunker/jsfiddle/codepen with the code?

Some news @outsourcegb @crutchcorn ?

**Comment by se-aurasys** [comment]
Haven't created plunker but I just copied code from equalTo validator and updated for newGte. Mainly 

`return v >= gtControl.value ? null : { newGte: true };`

line and it works perfectly.
